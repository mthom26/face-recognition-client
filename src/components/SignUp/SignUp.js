import React, { Component } from 'react';

import CustomForm from '../CustomForm/CustomForm';
import styles from './SignUp.module.css';

class SignUp extends Component {
  constructor(props) {
    super(props);
  }

  onSubmit = async formData => {
    try {
      const url = `${process.env.REACT_APP_SERVER_URL}/signup`;

      const result = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(formData),
        headers: {
          'Content-Type': 'application/json'
        }
      });

      const data = await result.json();
      this.props.setUser({
        id: data.id,
        name: data.name,
        email: data.email
      });
      this.props.setToken(data.token);
      console.log(data);
    } catch (err) {
      console.log(err);
    }
  };

  render() {
    return (
      <div className={styles.container}>
        <h2>Sign Up</h2>
        <CustomForm
          showName
          showEmail
          showPassword
          showPwConfirm
          onSubmitAction={this.onSubmit}
        />
      </div>
    );
  }
}

export default SignUp;
