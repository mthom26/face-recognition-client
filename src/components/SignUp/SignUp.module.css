.container {
  padding: 2rem;
  background: #00000076;
  display: flex;
  flex-direction: column;
  text-align: center;
}

.form {
  display: flex;
  flex-direction: column;
}

.button {
  color: #f2f2f2;
  margin: 1rem 0;
  padding: 0.8rem;
  opacity: 0.8;
  border: none;
  border-radius: 5px;
  background: linear-gradient(to right, #f3904f, #3b4371);
  transition: opacity 300ms ease-out;
}

.button:hover {
  opacity: 1;
}
