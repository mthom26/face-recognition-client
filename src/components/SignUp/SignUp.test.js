import React from 'react';
import { shallow } from 'enzyme';
import '../../setupEnzyme';

import SignUp from './SignUp';

describe('SignUp Component', () => {
  it('renders SignUp component', () => {
    expect(shallow(<SignUp />)).toMatchSnapshot();
  });
});
