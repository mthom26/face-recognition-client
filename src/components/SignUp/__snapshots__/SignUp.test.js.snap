// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders SignUp component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sign Up
        </h2>,
        <form>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
          <input
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <input
            id="confirmPassword"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
          <button
            onClick={[Function]}
          >
            Submit
          </button>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />,
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />,
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <input
              id="confirmPassword"
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />,
            <button
              onClick={[Function]}
            >
              Submit
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "name",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "password",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "confirmPassword",
              "name": "confirmPassword",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sign Up
          </h2>,
          <form>
            <input
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <input
              id="confirmPassword"
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
            <button
              onClick={[Function]}
            >
              Submit
            </button>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
                value=""
              />,
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />,
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              <input
                id="confirmPassword"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />,
              <button
                onClick={[Function]}
              >
                Submit
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "name",
                "name": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "confirmPassword",
                "name": "confirmPassword",
                "onChange": [Function],
                "placeholder": "Confirm Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
